src/oic/oic/provider.py

```diff
@@ -2049,3 +1882,471 @@ def get_by_sub_and_(self, sub: str, key: str, val: Any) -> Optional[str]:
+            "uid": uid,
+            "client_id": client_id,
+            "redirect_uri": redirect_uri,
+            "sid": sid,
+        }
+        if "state" in esr:
+            payload["state"] = esr["state"]
+
+        if self.events:
+            self.events.store("object args", "{}".format(payload))
+
+        # From me to me
+        _jws = JWT(
+            self.keyjar, iss=self.name, lifetime=86400, sign_alg=self.signing_alg
+        )
+        sjwt = _jws.pack(aud=[self.name], **payload)
+
+        location = "{}?{}".format(self.logout_verify_url, urlencode({"sjwt": sjwt}))
+        return SeeOther(location)
+
+    def unpack_signed_jwt(self, sjwt: str):
+        """Will unpack a signed JWT."""
+        verifier = JWT(self.keyjar)
+        try:
+            return verifier.unpack(sjwt)
+        except Exception as err:
+            raise ValueError(err)
+
+    def do_verified_logout(
+        self, sid: str, client_id: str, alla: bool = False, **kwargs
+    ) -> Dict:
+        """
+        Perform back channel logout and prepares the information needed for front channel logout.
+
+        :param sid: Session ID
+        :param client_id: Client ID
+        :param alla: Whether logout should be attempted from all clients or just one specific client.
+        :param kwargs:
+        :return:
+        """
+        if alla:
+            uid = self.sdb.get_uid_by_sid(sid)
+            logout_spec = self.logout_info_for_all_clients(uid)
+            # Find all the session IDs this user has gotten
+            sids = session_get(self.sdb, "uid", uid)
+        else:
+            logout_spec = self.logout_info_for_one_client(
+                session_id=sid, client_id=client_id
+            )
+            sids = [sid]
+
+        if self.events:
+            self.events.store("object args", "{}".format(logout_spec))
+
+        if not logout_spec["back_channel"] and not logout_spec["front_channel"]:
+            return {}
+
+        # take care of Back channel logout first
+        if logout_spec["back_channel"]:
+            failed = []
+            for _cid, spec in logout_spec["back_channel"].items():
+                _url, sjwt = spec
+                logger.info("logging out from {} at {}".format(_cid, _url))
+
+                try:
+                    res = self.httpc.http_request(
+                        _url, "POST", data="logout_token={}".format(sjwt)
+                    )
+                except Exception as err:
+                    # Can't be more specific because I don't know which http client are used
+                    logger.error("failed to logout from {}".format(_cid))
+                    if self.events:
+                        self.events.store("exception", "{}: {}".format(_cid, str(err)))
+                    failed.append(_cid)
+                    continue
+
+                if res.status_code < 300:
+                    logger.info("Logged out from {}".format(_cid))
+                elif res.status_code >= 400:
```

Comment:
It looks to me that this can be a simple `else` since redirects should be followed, aren't they?

Reply:
To assume that the client automatically follows a 30X is probably fair.